utils/interface.py

```diff
@@ -0,0 +1,256 @@
+
+    def process_data(self, now, line):
+        """ Save data in memory and on file storage
+
+        The first item is the computer time of reception as a datetime.datetime object
+        Data cannot be recorded if the header has not been received
+
+        Parameters
+        ----------
+        now : datetime object
+            time of reception of the line
+        line : string
+            line received from telemetry, without start and end character
+
+        """
+        if self.header:
+            data = [now] + line.split(self.separators['SEP_DATA'])
+            self.data.append(data)
+            self.write_data(data)
+
+    def write_calibration(self):
+        """ Write the calibration data in the file located at `self.calibration_path`
+
+        """
+        with open(self.calibration_path, 'w') as file:
+            file.write(yaml.dump(self.calibration))
+
+    def process_calibration(self, line):
+        """ Save calibration data in memory and on file storage
+
+        Parameters
+        ----------
+        line : string
+            line received from telemetry, without start and end character
+
+        """
+        # Don't save the calibration twice
+        if not self.calibration:
+            self.calibration = {value.split(self.separators['SEP_CALI'])[0]: value.split(
+                self.separators['SEP_CALI'])[1] for value in line.split(self.separators['SEP_DATA'])}
+            self.write_calibration()
+            print("Got calibration")
+
+    def write_message(self, message):
+        """ Append a line in the file located at `self.messages_path`
+
+        If the file does not exit, it is created
+
+        Parameters
+        ----------
+        message: string
+            message to write in the file
+
+        """
+        with open(self.messages_path, 'a+') as file:
+            file.write(message)
+
+    def process_message(self, now, line):
+        """ Save a message line in memory and on file storage
+
+        The message is saved in memory as a list [datetime.datetime, str]
+
+        The datetime.datetime object is stored as a string containing the time
+        in ISO format (YYYY-MM-DD HH:MM:SS.ffffff) on files storage
+        Use datetime.datetime.strptime(date_time_str, '%Y-%m-%d %H:%M:%S.%f') convert the time back to a 
+        datetime.datetime object
+
+        Parameters
+        ----------
+        now : datetime object
+            time of reception of the line
+        line : string
+            line received from telemetry, without start and end character
+
+        """
+        message = "{}\t{}\n".format(now, line)
+        self.messages.append([now, line])
+        self.write_message(message)
+        print("Message : {}".format("{} {}".format(now.time().replace(microsecond=0), line)))
+
+    def process_line(self, line):
+        if len(line) > 0:
+            line_start = line[0]
+            line_content = line[1:]
+            if line_start in self.separators_reversed:
```

Comment:
Should add a else: print('Line start error') or something

Reply:
Yes. All errors need to be better handled in general and saved in a file so we can keep track of them when we re-play the flight data afterward